learn/references/java-interoperability/the-bindgen-tool.md

```diff
@@ -0,0 +1,223 @@
+keywords: ballerina, programming language, java, interoperability, bindgen
+permalink: /learn/java-interoperability/the-bindgen-tool/
+active: the-bindgen-tool
+intro: The bindgen tool is a CLI tool, which generates Ballerina bindings for Java classes.
+redirect_from:
+  - /learn/tooling-guide/cli-tools/bindgen-tool/
+  - /learn/tooling-guide/cli-tools/bindgen-tool
+  - /learn/cli-documentation/bindgen-tool/
+  - /learn/cli-documentation/bindgen-tool
+  - /learn/cli-documentation/the-bindgen-tool/
+  - /learn/cli-documentation/the-bindgen-tool
+---
+
+The following sections explain how the `bindgen` tool works.
+
+## The `bindgen` Command
+
+```sh
+ballerina bindgen [(-cp|--classpath) <classpath>...]
+                  [(-mvn|--maven) <groupId>:<artifactId>:<version>]
+                  [(-o|--output) <output-path>]
+                  [--public]
+                  (<class-name>...)
+```
+
+`(-cp|--classpath) <classpath>...`
+This optional parameter could be used to specify one or more comma-delimited classpaths for retrieving the required Java libraries needed by the bindgen tool execution. The classpath could be provided as comma-separated paths of JAR files or as comma-separated paths of directories containing all the relevant Java libraries. If the Ballerina bindings are to be generated from a standard Java library, from a library available inside the Ballerina SDK, or from a platform library specified in the `Ballerina.toml`, then you need not specify the classpath explicitly.
+
+`(-mvn|--maven) <groupId>:<artifactId>:<version>`
+This optional parameter could be used to specify a Maven dependency required for the generation of the Ballerina bindings. Here, the specified library and its transitive dependencies will be resolved into the `target/platform-libs` directory of the package. If the tool is not executed inside a package or if the output path does not point to a package, the `target/platform-libs` directory structure will be created in the output path to store the Maven dependencies. The tool will also update the `Ballerina.toml` file with the platform libraries if the command is executed inside a Ballerina package.
+
+`(-o|--output) <output>`
+This optional parameter could be used to generate all the bindings inside a single directory instead of generating module level mappings. This option could be used in instances where all the mappings are required inside a single module. The specified directory doesn't always have to be inside a Ballerina package.
+
+`--public`
+Set the visibility modifier of the generated binding classes to public. This flag will be applicable only if the bindings are generated inside a single directory.
+
+`<class-name>...`
+One or more space-separated fully-qualified Java class names for which the Ballerina bridge code is to be generated. Please note that these class names should be provided at the end of the command.
```

Comment:
```suggestion
One or more space-separated fully-qualified Java class names for which the Ballerina bridge code is to be generated. Please note that these class names should be provided at the end of the command.
```
Add a  note.

Reply:
Since this should be a part of the option definition, adding a note would break the structure.